supervisor/utils/whoami.py

```diff
@@ -0,0 +1,58 @@
+"""Small wrapper for whoami API.
+
+https://github.com/home-assistant/whoami.home-assistant.io
+"""
+import asyncio
+from datetime import datetime
+import logging
+
+import aiohttp
+import attr
+
+from ..exceptions import WhoamiConnectivityError, WhoamiError, WhoamiSSLError
+from .dt import utc_from_timestamp
+
+_LOGGER: logging.Logger = logging.getLogger(__name__)
+_API_CALL: str = "whoami.home-assistant.io/v1"
+
+
+@attr.s(slots=True, frozen=True)
+class WhoamiData:
+    """Client Whoami data."""
+
+    timezone: str = attr.ib()
+    dt_utc: datetime = attr.ib()
+
+
+async def retrieve_whoami(
+    websession: aiohttp.ClientSession, with_ssl: bool = True
+) -> WhoamiData:
+    """Check if password is pwned."""
+    url: str = f"http{'s' if with_ssl else ''}://{_API_CALL}"
```

Comment:
Our Cloudflare rules rewrite all `http` requests to `https`, so `with_ssl` has no impact.